src/InteractiveWindow/Editor/History.cs

```diff
@@ -149,7 +149,13 @@ internal void Add(SnapshotSpan span)
 
         internal Entry GetNext(string pattern)
         {
-            return MoveNext(pattern);
+            var next = MoveNext(pattern);
+            if (next == null)
+            {
+                // if we hit the end of history list, reset _current to stop navigating history.
+                _current = -1;
+            }
+            return next;
```

Comment:
@ManishJayaswal This makes it parallel with ```GetPrevious```.  I assume both had it outside ```MoveX``` because there are multiple returns.  However, I agree that both could be inlined.